toolchain/check/context.h

```diff
@@ -265,8 +263,9 @@ class Context {
   // Prints a diagnostic for a duplicate name.
   auto DiagnoseDuplicateName(SemIRLoc dup_def, SemIRLoc prev_def) -> void;
 
-  // Prints a diagnostic for a poisoned name.
-  auto DiagnosePoisonedName(SemIRLoc loc) -> void;
+  // Prints a diagnostic for a poisoned name: Name used before declared.
+  auto DiagnosePoisonedName(SemIR::LocId poisoning_loc_id, SemIRLoc decl_loc)
```

Comment:
With all the discussion of `SemIRLoc`, is it needed here or can we use `InstId`? i.e.:
```suggestion
  auto DiagnosePoisonedName(SemIR::LocId poisoning_loc_id, SemIR::InstId decl_inst_id)
```

Reply:
Done.